jujugui/static/gui/src/app/components/deployment/deployment.js

```diff
@@ -60,11 +61,15 @@ YUI.add('deployment-component', function() {
         case 'add-credentials':
           return (
             <juju.components.DeploymentAddCredentials
-              changeState={this.props.changeState} />);
+              changeState={this.props.changeState}
+              deploymentStorage={this.props.deploymentStorage}
```

Comment:
I think I see what you're trying to do here, by passing the object down you're hoping to be able to modify the object by reference. This unfortunately isn't going to work, and I'm not really a fan of the 'side effects' approach. 

You'll have more luck if instead, you create a setter method in deployment.js which takes data and stores it in a property in the deployment component. You'd then pass this method into the child to call. You can then pass that deployment data into each component here along side the setter.